EU4toV2/Source/Mappers/CountryDetails/CountryDetails.cpp

```diff
@@ -0,0 +1,85 @@
+#include "CountryDetails.h"
+#include "ParserHelpers.h"
+#include "OSCompatibilityLayer.h"
+#include "../../Configuration.h"
+
+mappers::CountryDetails::CountryDetails(const std::string& _filename): filename(_filename)
+{
+	registerKeys();
+
+	if (Utils::DoesFileExist("./blankMod/output/common/countries/" + filename))
+	{
+		parseFile("./blankMod/output/common/countries/" + filename);
+	}
+	else if (Utils::DoesFileExist(theConfiguration.getVic2Path() + "/common/countries/" + filename))
+	{
+		parseFile(theConfiguration.getVic2Path() + "/common/countries/" + filename);
+	}
+	// Maybe we're initing a dead nation. If so look in the /other/ place.
```

Comment:
Wait, this can be initialized from either the common country file or the history? Don't they have completely separate types of information?

Reply:
Ugh, I just saw the original logic doing this. What a mess.

Reply:
hehehe.